mlflow/tracking/client.py

```diff
@@ -1729,6 +1729,58 @@ def log_table(
             # for data like {"inputs": "What is MLflow?"}
             except ValueError:
                 data = pd.DataFrame([data])
+
+        # Check if column is Image object and save filepath
+        if len(data.select_dtypes(include=["object"]).columns) > 0:
+            try:
+                import PIL
+            except ImportError as exc:
+                raise ImportError(
```

Comment:
This could lead to strange UX - for users not interested in image logging, they will still hit this code block. We should throw a warning and skip this part instead.